frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -520,10 +520,12 @@ def bitwise(
                 )
             self.error(highlights)
 
-        assert self.is_bit_width_compatible(resulting_type, x, y)
+        assert self.is_bit_width_compatible(x, y)
         assert resulting_type.is_encrypted and x.is_encrypted and y.is_encrypted
 
-        if x.original_bit_width + y.original_bit_width <= resulting_type.bit_width:
+        internal_result_type = max(x.type, y.type, key=lambda t: len(t.shape))
+
+        if x.original_bit_width + y.original_bit_width <= internal_result_type.bit_width:
```

Comment:
This is not good IMO, there will be cases where we could have used the optimized approach, but we can't because we're comparing it with `max(x.bit_width, y.bit_width)`

Reply:
This targets multi-parameters. Should we check here if we're multiparameter or not ?

Reply:
No need for that, it'll automatically work for single precision anyway, we can safely assume multi-precision and based the implementation on that.